qcodes/instrument_drivers/QuTech/S5i.py

```diff
@@ -20,34 +20,88 @@ class S5i(Instrument):
             individual modules.
 
         module (int): module number as set on the hardware.
+
+    Parameters:
+        enabled: switched output on/off
+        frequency: Frequency on Hz, range 40 MHz - 4 GHz, stepsize 1 MHz
+        freq_fine: Frequency on Hz, range 40 MHz - 4 GHz, stepsize 10 kHz.
+        Difference between frequency and freq_fine unclear!?
+        power: power in dBm, range -20 to +15 dBm
+
     """
 
     def __init__(self, name, spi_rack, module, **kwargs):
         super().__init__(name, **kwargs)
 
-        self.s5i = S5i_module(spi_rack, module)
-
-        self.add_parameter('use_external_reference',
-                           label='Use external reference',
-                           get_cmd=self._use_external_reference,
-                           set_cmd=self.s5i.use_external_reference,
+        self.s5i = S5i_module(spi_rack, module, frequency=41e6,
+                              enable_output=1, output_level=0)
+        # Put low freq as it is unlikely it will be resonant with something.
+        self.add_parameter('enabled',
+                           label='RF_on/off',
+                           set_cmd=self.s5i.enable_output_soft,
                            vals=Bool())
 
-        self.add_parameter('frequency_stepsize',
-                           label='Frequency stepsize',
-                           get_cmd=self._get_stepsize,
-                           set_cmd=self.s5i.set_stepsize,
-                           units='Hz',
-                           vals=Numbers())
-
         self.add_parameter('frequency',
                            label='Frequency',
                            get_cmd=self._get_rf_frequency,
                            set_cmd=self.s5i.set_frequency,
-                           units='Hz',
+                           unit='Hz',
                            vals=Numbers())
 
-        self.add_function('optimize_for_frequency', call_cmd=self._optimize)
+        self.add_parameter('freq_fine',
+                           label='Frequency fine',
+                           get_cmd=self._get_rf_frequency,
+                           set_cmd=self.s5i.set_frequency_optimally,
```

Comment:
just to confirm that my understanding is correct, is `s5i.set_frequency_optimally` just capable of setting the frequency with smaller steps or is there indeed some "optimization" involved? If the "optimization" is indeed involved, then I would recommend that the parameter be called differently (e.g. smth with "optimal"). 

Or, better, since the `frequency` parameter and `freq_fine` have the same "get" command, I would make just a public method out of `freq_fine` called `set_frequency_optimally` (or `set_frequency_finer`, or smth better) which will avoid the confusion about `freq_fine` being a separate/independent parameter which it is not.